ansible/collections/bento/common/roles/opensearch_dataloading/files/opensearch-backup.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+import boto3
+import requests
+from requests_aws4auth import AWS4Auth
+
+parser = argparse.ArgumentParser(description='Opensearch Backup Script')
+parser.add_argument("--oshost", type=str, help="opensearch host with trailing /")
+parser.add_argument("--repo", type=str, help="opensearch snapshot repository")
+parser.add_argument("--s3bucket", type=str, help="s3 bucket")
+parser.add_argument("--basePath", type=str, help="s3 bucket base path")
+parser.add_argument("--snapshot", type=str, help="opensearch snapshot value")
+parser.add_argument("--rolearn", type=str, help="role arn - typically power user role")
+args = parser.parse_args()
+print(args)
+oshost = args.oshost
+repo = args.repo
+
+s3bucket= args.s3bucket
+snapshot= args.snapshot 
+basepath= args.basePath
+rolearn= args.rolearn
+
+host = (oshost) 
+region = 'us-east-1' 
+service = 'es'
+credentials = boto3.Session().get_credentials()
+awsauth = AWS4Auth(credentials.access_key, credentials.secret_key, region, service, session_token=credentials.token)
+
+# Register repository
+headers = {"Content-Type": "application/json"}
+path = '_snapshot/'+repo
+url = host + path
+payload = {
+  "type": "s3",
+  "settings": {
+    "bucket": s3bucket,
+    "base_path": basepath,
+    "region": "us-east-1",
```

Comment:
should this reference the 'region' variable set earlier?

Reply:
region = boto3.Session().region_name